python/lsst/ap/pipe/metrics.py

```diff
@@ -0,0 +1,138 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+"""Metrics for ap_pipe tasks.
+"""
+
+__all__ = [
+    "ApFakesCompletenessMetricTask", "ApFakesCompletenessMetricConfig",
+]
+
+import astropy.units as u
+import numpy as np
+
+import lsst.pex.config as pexConfig
+from lsst.pipe.base import Struct
+import lsst.pipe.base.connectionTypes as connTypes
+from lsst.pipe.tasks.insertFakes import InsertFakesConfig
+from lsst.verify import Measurement, Name
+from lsst.verify.tasks import MetricTask, MetricComputationError
+
+
+class ApFakesCompletenessMetricConnections(
+        MetricTask.ConfigClass.ConnectionsClass,
+        dimensions={"instrument", "visit", "detector", "band"},
+        defaultTemplates={"coaddName": "deep",
+                          "fakesType": "",
+                          "package": "ap_pipe",
+                          "metric": "apFakesCompleteness"}):
+    """ApFakesCompleteness connections.
+    """
+    matchedFakes = connTypes.Input(
+        doc="Fakes matched to their detections in the difference image.",
+        name="{fakesType}{coaddName}Diff_matchDiaSrc",
+        storageClass="DataFrame",
+        dimensions=("instrument", "visit", "detector"),
+    )
+
+
+class ApFakesCompletenessMetricConfig(
+        MetricTask.ConfigClass,
+        InsertFakesConfig,
+        pipelineConnections=ApFakesCompletenessMetricConnections):
+    """ApFakesCompleteness config.
+    """
+    magMin = pexConfig.RangeField(
+        doc="Minimum magnitude the mag distribution. All magnitudes requested "
+            "are set to the same value.",
+        dtype=int,
+        default=20,
+        min=1,
+        max=40,
+    )
+    magMax = pexConfig.RangeField(
+        doc="Maximum magnitude the mag distribution. All magnitudes requested "
+            "are set to the same value.",
+        dtype=int,
+        default=30,
+        min=1,
+        max=40,
+    )
+
+    @property
+    def metricName(self):
+        """The metric calculated by a `MetricTask` with this config
+        (`lsst.verify.Name`, read-only).
+        """
+        return Name(package=self.connections.package,
+                    metric=f"{self.connections.metric}Mag{self.magMin}t{self.magMax}")
+
+
+class ApFakesCompletenessMetricTask(MetricTask):
+    """Metric task for summarizing the completeness of fakes inserted into the
+    AP pipeline.
+    """
+    _DefaultName = "apFakesCompleteness"
+    ConfigClass = ApFakesCompletenessMetricConfig
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        inputs = butlerQC.get(inputRefs)
+        inputs["band"] = butlerQC.quantum.dataId["band"]
+
+        outputs = self.run(**inputs)
+        butlerQC.put(outputs, outputRefs)
```

Comment:
Unfortunately, [`MetricTask.runQuantum`](https://github.com/lsst/verify/blob/master/python/lsst/verify/tasks/metricTask.py#L177-L189) has some error-handling logic that you've bypassed here; please add it in. Sorry about that; hopefully the extra code won't be necessary in the future.